sql/log_event_server.cc

```diff
@@ -7666,15 +7666,21 @@ int Rows_log_event::update_sequence()
   bool old_master= false;
   int err= 0;
 
-  if (!bitmap_is_set(table->rpl_write_set, MIN_VALUE_FIELD_NO) ||
-      (
-#if defined(WITH_WSREP)
-       ! WSREP(thd) &&
+  rpl_group_info *table_rgi=
+#ifdef WITH_WSREP
+  WSREP(thd) ? thd->wsrep_rgi :
 #endif
-       table->in_use->rgi_slave &&
-       !(table->in_use->rgi_slave->gtid_ev_flags2 & Gtid_log_event::FL_DDL) &&
+  table->in_use->rgi_slave;
+  rpl_group_info *thd_rgi=
+#ifdef WITH_WSREP
+  WSREP(thd) ? thd->wsrep_rgi :
```

Comment:
This is confusing, in line 7669 there is assignment to table_rgi and in line 7673 thd_rgi, why you need to set these both ?

Reply:
Maybe here IF_WSREP(A,B) macro could be clearer or even

#ifdef WITH_WSREP
if (WSREP(thd) {
  assigments;
 } else
#else
{
 assigments;
}
#endif